### ISSUE TITLE
regression: OSX: Library not loaded: @rpath/libdruntime-ldc-shared.78.dylib

### ISSUE THREAD

**Issue by timotheecour** [ldc-developers/ldc#2615]
/cc @klickverbot 

this used to work with lldb 1.6.0:

https://github.com/timotheecour/dtools/blob/master/dtools/lldbdplugin.d (demangle D in lldb)

ldmd2 -of=liblldbdplugin.dylib -shared -g -I$dtools_D $dtools_D/dtools/modified/demangle.d $dtools_D/dtools/lldbdplugin.d

$path_to/lldb -- $some_program

after upgrading ldc2 (via homebrew) to 1.8.0 it fails; after debugging this is because:
dlopen(/Users/timothee/git_clone/timotheecour/tims/tests/d31_gdb_demangle/liblldbdplugin.dylib, 9): Library not loaded: @rpath/libdruntime-ldc-shared.78.dylib
  Referenced from: /Users/timothee/homebrew/opt/ldc/lib/libphobos2-ldc-shared.78.dylib
  Reason: image not found

## root cause
```
otool -L liblldbdplugin.dylib
liblldbdplugin.dylib:
        liblldbdplugin.dylib (compatibility version 0.0.0, current version 0.0.0)
        /Users/timothee/homebrew/opt/ldc/lib/libphobos2-ldc-shared.78.dylib (compatibility version 78.0.0, current version 2.0.78)
        /Users/timothee/homebrew/opt/ldc/lib/libdruntime-ldc-shared.78.dylib (compatibility version 78.0.0, current version 2.0.78)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)

otool -L /Users/timothee/homebrew/opt/ldc/lib/libphobos2-ldc-shared.78.dylib
/Users/timothee/homebrew/opt/ldc/lib/libphobos2-ldc-shared.78.dylib:
        /Users/timothee/homebrew/opt/ldc/lib/libphobos2-ldc-shared.78.dylib (compatibility version 78.0.0, current version 2.0.78)
        @rpath/libdruntime-ldc-shared.78.dylib (compatibility version 78.0.0, current version 2.0.78)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)

#=>  see @rpath/
```

whereas with ldc 1.6.0:
```
otool -L liblldbdplugin.dylib
liblldbdplugin.dylib:
        liblldbdplugin.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

**Comment by timotheecour** [comment]
workaround:
adding this to cmd line seems to work:
```
-Xcc=-Wl,-rpath,/Users/timothee/homebrew/opt/ldc/lib/
```
problem is, that's not DRY as we need to 'guess' that path ; in more complex situations (eg multiple versions of ldc) this could be problematic.

**Comment by kinke** [comment]
As the release notes say, produced shared libs via `-shared` now link against shared druntime/Phobos by default, so setting the rpath during linking is now required (unless reverting to `-link-defaultlib-shared=false`). To [quote yourself](https://github.com/ldc-developers/ldc/issues/2442#issuecomment-351326872):
> @rpath/{druntime,phobos2}-ldc-shared.dylib + telling users to set -rpath accordingly seems indeed like the preferred way to go.

**Comment by kinke** [comment]
[The rpath can also be set in the config file via something like `-Xcc=-Wl,-rpath,%%ldcbinarypath%%/../lib`.]

I don't know where the inconsistency comes from. Can you please check the official release package too? [I have no Mac.]

**Comment by timotheecour** [comment]
build from ldc master:
works fine even without `-Xcc=-Wl,-rpath,/Users/timothee/homebrew/opt/ldc/lib/` or `-link-defaultlib-shared=false` because bin/ldc2.conf is:

```


  // See comments in driver/config.d in ldc source tree for grammar description of
  // this config file.

  // The default group is required
  default:
  {
      // default switches injected before all explicit command-line switches
      switches = [
          "-I/Users/timothee/git_clone/D/ldc2/runtime/druntime/src",
          "-I/Users/timothee/git_clone/D/ldc2/runtime/jit-rt/d",
          "-I/Users/timothee/git_clone/D/ldc2/runtime/phobos",
          "-L-Wl,-rpath,/Users/timothee/git_clone/D/ldc2/build/lib",
          "-defaultlib=phobos2-ldc,druntime-ldc"
      ];
      // default switches appended after all explicit command-line switches
      post-switches = [
          "-L-L/Users/timothee/git_clone/D/ldc2/build/lib",
      ];
  };
```
which contains rpath


however in homebrew we have:
$homebrew_D/Cellar/ldc/1.8.0_2/etc/ldc2.conf

```
// See comments in driver/config.d in ldc source tree for grammar description of
  // this config file.

  // The default group is required
  default:
  {
      // default switches injected before all explicit command-line switches
      switches = [
          "-I/Users/timothee/homebrew/Cellar/ldc/1.8.0_2/include/dlang/ldc/ldc",
          "-I/Users/timothee/homebrew/Cellar/ldc/1.8.0_2/include/dlang/ldc",
          "-defaultlib=phobos2-ldc,druntime-ldc"
      ];
      // default switches appended after all explicit command-line switches
      post-switches = [
          "-L-L/Users/timothee/homebrew/Cellar/ldc/1.8.0_2/lib",
      ];
  };

```
which doesn't contain rpath, hence the error

so looks like homebrew ldc.rb is the culprit

**Comment by timotheecour** [comment]
see PR: https://github.com/ldc-developers/ldc/pull/2616

**Comment by kinke** [comment]
Fixed by #2659.
